scourgify/normalize.py

```diff
@@ -491,11 +502,15 @@ def normalize_directionals(parsed_addr):
             dir_str, exclude=[], removal_cats=STRIP_ALL_CATS, strip_spaces=True
         )
         if dir_str in DIRECTIONAL_REPLACEMENTS.keys():
-            parsed_addr[found] = DIRECTIONAL_REPLACEMENTS[dir_str]
+            dir_str = DIRECTIONAL_REPLACEMENTS[dir_str]
+        if long_hand:
```

Comment:
I'm not sure I'm following your train of thought here, or what you mean by "Passing in long_hand from the calling libraries".
This package has always only supported normalizing addresses with their directional and street type abbreviations.  This change provides the _option_ to have those elements in long hand instead of abbreviation.  So to me it tracks that yes you would need to know which version you are using, but ONLY if you desire to have long hand addresses. 
It's a feature upgrade that provides a new option, but does NOT change the original, expected behavior unless expressly requested.
The upgrade provides the option to receive the normalized address with long form directionals and street types, ONLY if desired, by passing the `long_hand` parameter to the `NormalizeAddress` class instantiation or as a parameter to the normalization functions.  This parameter defaults to `False` which preserves the original behavior of the package.
Having a master collection of entries is problematic, particularly for street types, for reasons described in my answer to your question below.  
Additionally, the package is designed to allow a wide range of overrides, so having separate objects for each replacement mapping makes it easier to override, or update, just the desired mappings, or portions of mappings.

Reply:
My point on this is that by requiring the calling library to know ahead of time whether or not an address uses abbreviations, we are limiting the ability to handle various types of input data. Maybe there's an upstream process in GBR or other apps that normalizes all inputs to either long hand or abbreviations. If so, this is probably OK, but it would be nice to have _all_ address manipulation in one library, rather than scattered. So for example, if a data source not under our control (say a CSV import) suddenly switched from abbreviations to long hand, or even a mix of the two, scourgify would still be able to parse them without throwing errors.

Reply:
Ah.  I think I see the problem.  Which mostly comes down to me not updating the documentation.
The `long_hand` parameter is about OUTPUT not input.
So say I get the following addresses:
```
123 SW West Circ
123 SouthWest West Circle
123 south-west west crcl
```
The expected output of every one of these for `normalize_address_record(address)` or `NormalizeAddress(address).normalize()` would be: `123 SW WEST CIR`. 

This is both how the package has always behaved, and how it behaves moving forward.

But the expected output of every one of these for `normalize_address_record(address, long_hand=True)` or `NormalizeAddress(address, long_hand=True).normalize()` would be: `123 SOUTHWEST WEST CIRCLE`, because the new `long_hand` parameter provides the option to request the OUTPUT be in a long hand format, rather than the default abbreviated.

So in that you do have to know what version number of the package you are using, IF you want to use the long form output.  If you want to use the original, and still default, behavior, then the package version isn't an issue.